### PR TITLE
feat(events): Fetch from /api/0/organizations/:org/events-stats/ with multiple yAxis options

### PR DESCRIPTION
**Pull request by dashed**
Update some frontend code that fetches from `/api/0/organizations/:org/events-stats/` to be able to pass multiple `yAxis` options, and parse the response that contains sets of chart data.

The backend changeset that adds this was in https://github.com/getsentry/sentry/pull/17177

The Performance page will make use of this.

------

For the TypeScript pros, this PR also adds `assertType<Type>` type utility function to assert types without resorting to generic types gymnastics.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx [hunk 1]
```diff
@@ -15,21 +16,27 @@ import SentryTypes from 'app/sentryTypes';
...
-  reloading: boolean;
-  errored: boolean;
-
+type TimeSeriesData = {
   // timeseries data
   timeseriesData?: Series[];
   allTimeseriesData?: EventsStatsData;
   originalTimeseriesData?: EventsStatsData;
-  timeseriesTotals?: object;
+  timeseriesTotals?: {count: number};
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx]: <3

File: src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx [hunk 2]
```diff
@@ -15,21 +16,27 @@ import SentryTypes from 'app/sentryTypes';
...
   originalPreviousTimeseriesData?: EventsStatsData | null;
   previousTimeseriesData?: Series | null;
   timeAggregatedData?: Series | {};
 };
 
+type LoadingProps = {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx]: I think I'd give it another name.... `LoadStatus` for example

**Review by priscilawebdev — APPROVED** [review]
Good Job! It looks good to me 👍

File: src/sentry/static/sentry/app/types/utils.tsx [hunk 3]
```diff
@@ -6,3 +6,8 @@
 // value is true:
 // eslint-disable-next-line prettier/prettier
 export function assert(_value: unknown): asserts _value {}
+
+// This declares a function which asserts that the expression called
+// value is of type Type:
+// eslint-disable-next-line prettier/prettier
+export function assertType<Type>(_value: unknown): asserts _value is Type {}
```
[the remark this answers is not in the document]
Comment (dashed) [reply src/sentry/static/sentry/app/types/utils.tsx]: As far as I understand the way `asserts` works in TS is that It doesn't fail.

> It ensures that whatever condition is being checked must be true for the remainder of the containing scope.

My understanding is based on https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions

Same for the `assert()` function.



It's meant for type refinements without having to do a variable reassignment just for type casting. e.g. `value: T = value` or `value = value as T`

File: src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx (hunk 2 above)
Comment (dashed) [reply src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx]: Thanks. I'll rename this.

Comment (dashed) [reply src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx]: Done in https://github.com/getsentry/sentry/pull/17334/commits/88cb0f82f1a9b9164c72bdec6bf7be21e1ef01ea

File: src/sentry/static/sentry/app/types/utils.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (markstory) [reply src/sentry/static/sentry/app/types/utils.tsx]: Sure but won't this assertion allow usage like `assertType<string>(12)` pass when they probably shouldn't.

[the remark this answers is not in the document]
Comment (dashed) [reply src/sentry/static/sentry/app/types/utils.tsx]: Yea it does. But that's not really useful. These functions exist to compensate for the TS compiler not being smart; so we have to give them hints.

**Pull request merged**